### PR TITLE
[SDPA-1225] Summary-field-ristricted-to-200-characters

### PR DESCRIPTION
**Pull request by vincent-gao**
More details please see: https://github.com/dpc-sdp/content-vic-gov-au/pull/604
jira: https://digital-engagement.atlassian.net/projects/SDPA/issues/SDPA-1225

### CODE REVIEW

**Comment by anthony-malkoun** [comment]
@sonnykt may I have an opinion on this please. Should `tide_landing_page` have a `hook_update_N` to update the configuration do you think?

File: config/install/core.entity_form_display.paragraph.card_event.default.yml [hunk 1]
```diff
@@ -61,9 +62,14 @@ content:
     type: string_textarea
     weight: 5
     settings:
-      placeholder: ''
       rows: 5
-    third_party_settings: {  }
+      placeholder: ''
+    third_party_settings:
+      maxlength:
```
Comment (sonnykt) [inline config/install/core.entity_form_display.paragraph.card_event.default.yml]: @vincent-gao we also need a hook_update to update this config change. Please make sure that the hook_update only changes the `third_party_settings` of the field if `maxlength` hasn't been set.

Comment (vincent-gao) [reply config/install/core.entity_form_display.paragraph.card_event.default.yml]: added a hook_update to `tide_landing_page.install` file.

File: tide_landing_page.install [hunk 2]
```diff
@@ -384,3 +384,22 @@ function tide_landing_page_update_8008() {
...
+/**
+ * Summary field in carousal Card event is restricted to 200 characters.
+ */
+function tide_landing_page_update_8009() {
+  /** @var \Drupal\Core\Entity\Display\EntityFormDisplayInterface $entity_form_display */
+  $entity_form_display = Drupal::entityTypeManager()
+    ->getStorage('entity_form_display')
+    ->load('paragraph.card_event.default');
+  $field_paragraph_summary = $entity_form_display->getComponent('field_paragraph_summary');
+  if (!$field_paragraph_summary['third_party_settings']) {
```
Comment (GROwen) [inline tide_landing_page.install]: @vincent-gao I think if `['maxlength']['maxlength_js']` is set there shouldn't be any forced change, so I suggest extending this conditional statement to perform a check of adding a check in this conditional for `['maxlength']['maxlength_js']` and removing the `elseif`

Comment (vincent-gao) [reply tide_landing_page.install]: thanks @GROwen , I added logic for checking if the `maxlength_js` is empty. if yes, i will force `maxlength_js` to 200.

File: tide_landing_page.install [hunk 3]
```diff
@@ -384,3 +384,18 @@ function tide_landing_page_update_8008() {
...
+/**
+ * Summary field in carousal Card event is restricted to 200 characters.
+ */
+function tide_landing_page_update_8009() {
+  /** @var \Drupal\Core\Entity\Display\EntityFormDisplayInterface $entity_form_display */
+  $entity_form_display = Drupal::entityTypeManager()
+    ->getStorage('entity_form_display')
+    ->load('paragraph.card_event.default');
+  $field_paragraph_summary = $entity_form_display->getComponent('field_paragraph_summary');
+  if (!$field_paragraph_summary['third_party_settings'] || !$field_paragraph_summary['third_party_settings']['maxlength']['maxlength_js']) {
```
Comment (sonnykt) [inline tide_landing_page.install]: @vincent-gao you need to check with either `isset()` or `!empty()`. Checking `!$var` will result in a PHP warning if `$var` is not set.

The hook_update sets the maxlength settings for existing fields if they are *not set*. If the field already has `$field_paragraph_summary['third_party_settings']['maxlength']`, the hook_update should not overwrite the settings. Your hook_update also needs to set all the settings of maxlength as per the config, not just `maxlength_js`.

Comment (vincent-gao) [reply tide_landing_page.install]: Thanks, Sonny. if `maxlength_js` already has a value, we don't want to change it. 
fixed other requirements.

**Pull request closed without merging**